lib/internal/Magento/Framework/Url.php

```diff
@@ -478,8 +488,7 @@ public function getBaseUrl($params = [])
         ) {
             $this->getRouteParamsResolver()->setType(UrlInterface::URL_TYPE_DIRECT_LINK);
         }
-
-        $result = $this->_getScope()->getBaseUrl($this->_getType(), $this->_isSecure());
+        $result = self::_getScope()->getBaseUrl($this->_getType(), $this->_isSecure());
```

Comment:
I don't think this change from `$this->_getScope()` to `self::_getScope()` is needed. Can you please provide more details on why do you think it is necessary

Reply:
@sivaschenko   with `self::` we get ` _getScope()` in Framework/Url 
![deepin-screen-recorder_Select area_20190425145526](https://user-images.githubusercontent.com/2481206/56734074-937abe80-676a-11e9-8053-8aa1484b4f36.gif)
2 - with `$this ` we get ` _getScope()` in Model/Url wich returns admin scope.
![deepin-screen-recorder_Select area_20190425145648](https://user-images.githubusercontent.com/2481206/56734110-a9887f00-676a-11e9-8697-a7b110fd37bb.gif)

Reply:
But this is not good idea to use one method, when we need to call product url in admin scope, i thinks would be better to use pervious solution with **$this->getData('_scope')** and `if `statement, because some tests will fails with this

Reply:
I think this solution will be better and not break any functional
`$result  = $this->setRequestScope($params)->getBaseUrl($this->_getType(), $this->_isSecure());`
 
```
    private function setRequestScope($params)
    {
        $customScope = $this->getRouteParamsResolver()->getRouteParam('_scope');

        if (isset($params['_scope']) ) {
            $this->setScope($params['_scope']);
        } else if ($customScope) {
            $this->setScope($customScope);
            return $this->getData('scope');
        }
        return $this->_getScope();
    }
```